packages/git/src/browser/git-resource.ts

```diff
@@ -17,10 +17,11 @@ export class GitResource implements Resource {
 
     constructor(readonly uri: URI, protected readonly repository: Repository | undefined, protected readonly git: Git) { }
 
-    async readContents(options?: { encoding?: string | undefined; } | undefined): Promise<string> {
+    async readContents(options?: { encoding?: string }): Promise<string> {
         if (this.repository) {
             const commitish = this.uri.query;
-            return await this.git.show(this.repository, this.uri.toString(), Object.assign({ commitish }, options));
+            const content = await this.git.show(this.repository, this.uri.toString(), Object.assign({ commitish }, options));
+            return content === undefined ? '' : content;
```

Comment:
I've just tested that change, because I assume it has a implication on dirty diff manager (on master), but I was surprised by the value of `content`, it's `null`.

Reply:
Q: is it true, that `readContents` requires a result, or may it reject? 
if `readContents` may not reject, that implies a follow-up change, I kind of used that in dirty diff manager to detect that situation. if that `null` thing is sorted out, the previous revision state for dirty diff computation would be an empty string, witch would mark all contents of git-ingored files as added :-(

I think it's not a big deal, instead of using the git resource, we'd need to call `git.show` directly and check the result.

Reply:
What is your preference? If it can reject, we have to cover it all over the application. I would go with calling `git show` directly. What's your take on that?

Reply:
I searched for references of `Resource.readContents` and it looks like otherwise it's never considered it might resolve with `undefined`.

calling `git show` would fix that too. if you like, I'll create PR for yours.